compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/PsiUtils.kt

```diff
@@ -474,7 +474,11 @@ public fun KtCallableDeclaration.requireTypeReference(module: ModuleDescriptor):
     )
   }
 
-  throw AnvilCompilationException("Couldn't obtain type reference.", element = this)
+  throw AnvilCompilationException(
+    message = "Couldn't obtain type reference. Did you declare a type for ${this.name} " +
+      "explicitly?",
```

Comment:
Just a suggestion that I think is a bit easier to grok at a glance

```suggestion
    message = "Couldn't obtain type reference. Did you declare an explicit type for ${this.name}?",
```